parser/src/parser/tests/mod.rs

```diff
@@ -1,18 +1,619 @@
+            span: miden_diagnostics::SourceSpan::UNKNOWN,
+            name: ResolvableIdentifier::Global(ident!($name)),
+            access_type: AccessType::Default,
+            offset: 0,
+            ty: Some($ty),
+        })
+    };
+
+    ($name:literal, $ty:expr) => {
+        ScalarExpr::SymbolAccess(SymbolAccess {
+            span: miden_diagnostics::SourceSpan::UNKNOWN,
+            name: ResolvableIdentifier::Global(ident!($name)),
+            access_type: AccessType::Default,
+            offset: 0,
+            ty: Some($ty),
+        })
+    };
+}
+
+macro_rules! access {
+    ($name:ident) => {
+        ScalarExpr::SymbolAccess(SymbolAccess::new(
+            miden_diagnostics::SourceSpan::UNKNOWN,
+            ident!($name),
+            AccessType::Default,
+            0,
+        ))
+    };
+
+    ($name:literal) => {
+        ScalarExpr::SymbolAccess(SymbolAccess::new(
+            miden_diagnostics::SourceSpan::UNKNOWN,
+            ident!($name),
+            AccessType::Default,
+            0,
+        ))
+    };
+
+    ($module:ident, $name:ident, $ty:expr) => {
+        ScalarExpr::SymbolAccess(SymbolAccess {
+            span: miden_diagnostics::SourceSpan::UNKNOWN,
+            name: ResolvableIdentifier::Resolved(ident!($module, $name)),
+            access_type: AccessType::Default,
+            offset: 0,
+            ty: Some($ty),
+        })
+    };
+
+    ($name:ident, $offset:literal) => {
+        ScalarExpr::SymbolAccess(SymbolAccess::new(
+            miden_diagnostics::SourceSpan::UNKNOWN,
+            ident!($name),
+            AccessType::Default,
+            $offset,
+        ))
+    };
+
+    ($name:literal, $offset:literal) => {
+        ScalarExpr::SymbolAccess(SymbolAccess::new(
+            miden_diagnostics::SourceSpan::UNKNOWN,
+            ident!($name),
+            AccessType::Default,
+            $offset,
+        ))
+    };
+
+    ($name:ident, $offset:literal, $ty:expr) => {
+        ScalarExpr::SymbolAccess(SymbolAccess {
+            span: miden_diagnostics::SourceSpan::UNKNOWN,
+            name: ResolvableIdentifier::Local(ident!($name)),
+            access_type: AccessType::Default,
+            offset: $offset,
+            ty: Some($ty),
+        })
+    };
+
+    ($name:ident, $ty:expr) => {
+        ScalarExpr::SymbolAccess(SymbolAccess {
+            span: miden_diagnostics::SourceSpan::UNKNOWN,
+            name: ResolvableIdentifier::Local(ident!($name)),
+            access_type: AccessType::Default,
+            offset: 0,
+            ty: Some($ty),
+        })
+    };
+
+    ($module:ident, $name:ident, $offset:literal, $ty:expr) => {
+        ScalarExpr::SymbolAccess(SymbolAccess {
+            span: miden_diagnostics::SourceSpan::UNKNOWN,
+            name: ResolvableIdentifier::Resolved(ident!($module, $name)),
+            access_type: AccessType::Default,
+            offset: $offset,
+            ty: Some($ty),
+        })
+    };
```

Comment:
Do module and offset make sense in the same case, since offsets can only be applied to trace columns and those are always declared in the root module? (or is this related to handling of the evaluators? I haven't gone through all of the semantic analysis work yet)

Reply:
Yes this case can go away